pkg/mcproducer/scripts/replace-kernel-module.sh

```diff
@@ -19,17 +21,22 @@ EOF
     cat "$kmm_config_file_filepath"
 }
 
-echo "before checking image tar file presence"
-if [ -e $kernel_module_image_filepath ]; then
-    echo "Image file $kernel_module_image_filepath found on the local file system, creating kmm config file"
+echo "before checking image presence"
+if [ -n "$(podman images -q $worker_image 2> /dev/null)" ]; then
+    echo "Image $worker_image found on the local file system, creating kmm config file"
     create_kmm_config
-    echo "running kernel-management worker image"
-    podman run --user=root --privileged -v /lib/modules:/lib/modules -v $kmm_config_file_filepath:/etc/kmm-worker/config.yaml -v $kernel_module_image_filepath:$kernel_module_image_filepath $worker_image kmod load --tarball /etc/kmm-worker/config.yaml
+    podman pod create --name kmm-pod
+    echo "running init container image"
+    init_container_id=$(podman create --pod kmm-pod --user=root --init-ctr=once --privileged -v /tmp:/tmp $full_kernel_module_image /bin/sh -c "mkdir -p /tmp/opt/lib/modules; cp -R /opt/lib/modules/* /tmp/opt/lib/modules;")
+    podman start "$init_container_id"
```

Comment:
question: do we need to start each container in the pod, or can we just start the pod, and it will know how to deal with its containers?

Reply:
Thought about it, the only solution I found is to run a YAML.
for example:
```
podman play kube <<EOF
apiVersion: v1
kind: Pod
metadata:
  name: my-pod
spec:
  initContainers:
  - name: init-container
    image: init-contaier-image
  containers:
  - name: container
    image: container-image
EOF
```
But I thought it will be less clean in the code itself

Reply:
I am not sure i understand why we need "podman play kube". You have just created a pod and added 2 containers to ut. Can't we just run "podman pod start <pod name>"?

Reply:
So `podman` does support running a pod with an init-container eventually?
I definitely prefer applying a yaml - it will be much more similar to what we have in day2.

Reply:
we don't apply a yaml in day2: we simply create a pod with init-container and worker and start it

Reply:
But if we can apply a yaml using `podman` won't it be cleaner? Instead of having a long podman command mounting everything in the same command?